TUTORIAL.md

```diff
@@ -265,53 +309,26 @@ Ref String
 
 The last kind of type which gluon has is the alias type. An alias type is a type which explicitly names some underlying type which can either be one of the three types mentioned above or an abstract type which is the case for the `Int`, `String` and `Ref` types. If the underlying type is abstract then the type is only considered equivalent to its own alias (ie if you define an abstract type of `MyInt` which has the same representation as `Int` the typechecker still considers these two types as being not equal to each other).
 
-#### Higher-kinded types (TODO)
+### Higher-kinded types (TODO)
 
 Gluon has higher kinded types.
 
-### Indentation
+### Overloading
 
-If you have been following along this far you may be thinking think that the syntax so far is pretty limiting. In particular you wouldn't be wrong in thinking that the `let` and `type` syntax are clunky due to their need to be closed by the `in` keyword. Luckily gluon offers a more convenient way of writing bindings by relying on indentation.
+Sometimes there there is a need to overload a name with multiple differing implementations and let the compiler chose the correct implementation depending on which the inferred type is. If you have written any amount of Gluon code so far you are likely to have already encountered this with numeric operators such as `(+)` or comparison operators such as `(==)`. While these operators are core parts of gluon they are not special cased by the compiler which lets you define and use overloaded bindings yourself.
 
-When a token starts on the same column as an unclosed `let` or `type` expression the lexer implicitly inserts an `in` token which closes the declaration part and makes the following expression into the body.
+To explain how overloading works first look at the example below where `showType` has two implementations, one which takes an `Int` and one which takes a `Float`.
 
 ```f#,rust
-let add1 x = x + 1
-add1 11 // `in` will be inserted automatically since `add1 11` starts on the same line as the opening `let`
-```
+let showType _: Int -> String = "Int"
+let showType _: Float -> String = "Float"
 
-If a token starts on the same column as an earlier expression but there is not an unclosed `type` or `let` expression gluon treats the code as a block expression which means each expression are run sequentially, returning the value of the last expression.
-
-```f#
-do_something1 ()
-do_something2 () // `do_something1 ()` is run, then `do_something_2`. The result of `type ...` is the result of the expression
-type PrivateType = | Private Int
-let x = Private (do_something3 ())
-do_something3 ()
-match x with
-| Private y -> do_something4 x
+showType 0 // Returns "Int"
+showType 0.0 // Returns "Float"
+// showType "" // Would be a type error
 ```
```

Comment:
What casing are we using? This is camel, but most are snake case.

Also, spacing between the type and colon. It would be nice to have consistency (I prefer a space between, but eh).